website/docs/r/vapp_nat_rules.html.markdown

```diff
@@ -0,0 +1,163 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_vapp_nat_rules"
+sidebar_current: "docs-vcd-resource-vapp-nat-rules"
+description: |-
+  Provides a vCloud Director vApp NAT resource. This can be used to create, modify, and delete NAT rules.
+---
+
+# vcd\_vapp\_nat\_rules
+
+Provides a vCloud Director vApp NAT resource. This can be used to create,
+modify, and delete NAT rules in a [vApp network](/docs/providers/vcd/r/vapp_network.html).
+
+!> **Warning:** Using this resource overrides any existing NAT rules on vApp network. It's recommended to have only one resource per vApp. 
```

Comment:
Maybe there's a way to enforce a single `vcd_vapp_nat_rules` resource with the same `vapp_id` and `network_id`? Right now, if you create two resources, they overwrite each other's rules with each `terraform apply` (example config below).

> It's recommended to have only one resource per vApp. 

Is it really per vApp or per vApp + same vApp Network?

```
# NAT rules in the vApp Network
resource "vcd_vapp_nat_rules" "vapp_nat" {
  vapp_id              = vcd_vapp.demo_vapp.id
  network_id           = vcd_vapp_network.demo_vapp_net_w_org.id
  nat_type             = "portForwarding"
  enable_ip_masquerade = false

  rule {
    external_port   = "22"
    forward_to_port = "-1"
    #protocol        = "TCP_UDP"
    protocol        = "TCP"
    vm_nic_id       = "0"
    vm_id           = vcd_vapp_vm.demo_vm_wordpress.id
  }

  rule {
    external_port   = "-1"
    forward_to_port = "80"
    protocol        = "TCP"
    vm_nic_id       = "0"
    vm_id           = vcd_vapp_vm.demo_vm_wordpress.id
  }
}

# NAT rules in the vApp Network
resource "vcd_vapp_nat_rules" "vapp_nat_2" {
  vapp_id              = vcd_vapp.demo_vapp.id
  network_id           = vcd_vapp_network.demo_vapp_net_w_org.id
  nat_type             = "portForwarding"
  enable_ip_masquerade = false

  rule {
    external_port   = "443"
    forward_to_port = "4443"
    protocol        = "TCP"
    vm_nic_id       = "0"
    vm_id           = vcd_vapp_vm.demo_vm_wordpress.id
  }
}
```

Reply:
our provider don't manage resources and can't access other resources state or should do that.
> Is it really per vApp or per vApp + same vApp Network?
good point vapp+network

Reply:
description updated